File: Leetcode 2110 Number of Smooth Descent Periods of a Stock.py
# 2025/12/15
class Solution:
    def getDescentPeriods(self, prices: List[int]) -> int:
        # slide window
        n = len(prices)
        if n < 2:
            return 1
        ans = 1
        curr = 1
        for i in range(1, n):
            if prices[i-1] - prices[i]==1:
                curr+=1
            else:
                curr=1
            ans+=curr
        return ans

File: Leetcode 2110 Number of Smooth Descent Periods of a Stock.go
// 2025/12/15

func getDescentPeriods(prices []int) int64 {
    if len(prices) < 2{
        return 1
    }
    var ans int64 = 1
    var curr int64 = 1
    
    for i:=1; i < len(prices); i++{
        if prices[i-1] - prices[i]==1{
            curr += 1
        } else {
            curr = 1
        }
        ans += curr
    }
    return ans
}
